src/lsdb/catalog/dataset/healpix_dataset.py

```diff
@@ -205,41 +206,104 @@ def _get_non_empty_partitions(self) -> Tuple[List[HealpixPixel], List[Delayed]]:
         return non_empty_pixels, non_empty_partitions
 
     def skymap_data(
-        self, func: Callable[[pd.DataFrame, HealpixPixel], Any], **kwargs
+        self, func: Callable[[pd.DataFrame, HealpixPixel], Any], order: int | None = None, **kwargs
     ) -> Dict[HealpixPixel, Delayed]:
         """Perform a function on each partition of the catalog, returning a dict of values for each pixel.
 
         Args:
             func (Callable[[pd.DataFrame, HealpixPixel], Any]): A function that takes a pandas
                 DataFrame with the data in a partition, the HealpixPixel of the partition, and any other
                 keyword arguments and returns an aggregated value
+            order (int | None): The HEALPix order to compute the skymap at. If None (default), will compute
+                for each partition in the catalog at their own orders
             **kwargs: Arguments to pass to the function
 
         Returns:
             A dict of Delayed values, one for the function applied to each partition of the catalog
         """
 
         partitions = self.to_delayed()
-        results = {
-            pixel: delayed(func)(partitions[index], pixel, **kwargs)
-            for pixel, index in self._ddf_pixel_map.items()
-        }
+        if order is None:
+            results = {
+                pixel: delayed(func)(partitions[index], pixel, **kwargs)
+                for pixel, index in self._ddf_pixel_map.items()
+            }
+        else:
+            if order < self.hc_structure.pixel_tree.get_max_depth():
```

Comment:
I do not understand if it's useful to allow for fixed orders higher than the catalog's maximum order. In practice, won't all those plots look the same as if they were computed at the maximum order?